core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -128,14 +103,16 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
     Topic.prototype.deleteSubtopic = function(subtopicId, isNewlyCreated) {
       var subtopicDeleted = false;
       for (var i = 0; i < this._subtopics.length; i++) {
-        if ((this._subtopics[i].getId() === subtopicId)) {
+        if (this._subtopics[i].getId() === subtopicId) {
           // When a subtopic is deleted, all the skills in it are moved to
           // uncategorized skill ids.
-          var skillIds = this._subtopics[i].getSkillIds();
-          for (var j = 0; j < skillIds.length; j++) {
-            var skillId = skillIds[j];
-            if (this._uncategorizedSkillIds.indexOf(skillId) === -1) {
-              this._uncategorizedSkillIds.push(skillId);
+          var skills = this._subtopics[i].getSkillSummaries();
+          for (var j = 0; j < skills.length; j++) {
+            var skillId = skills[j].getId();
+            var skillDescription = skills[j].getDescription();
+            if (!this.hasUncategorizedSkill(skillId)) {
+              this._uncategorizedSkillSummaries.push(
+                SkillSummaryObjectFactory.create(skillId, skillDescription));
```

Comment:
Any reason not to just push the skill summary object directly?

Reply:
Not really, but if for some reason, that skill was there in the uncategorized section and the subtopic, then deleting that subtopic shouldn't create 2 copies of that skill in the uncategorized section. This case wouldn't happen currently, but just wrote that as an additional check.

Reply:
Hm ok -- I'm not too concerned either way. The real question is whether it matters to break references or not; having one instance of a summary of a given skill and using that throughout seemed reasonable to me.

Reply:
Yeah, when any move operation happens, the skill is deleted from source and added to destination, so this case should never happen. This is just as an additional check.